### PR TITLE
Feature/upgrade php and phpunit

### LINKED ISSUE #33: PHP Warnings due to missing defaults in CatalogCommand
### Describe the bug

There a few PHP Warnings in the plugin when running on PHP 8.2.

```log
Deprecated: Optional parameter $sites declared before required parameter $opts is implicitly treated as a required parameter in /chroot/var/www/wp-content/plugins/block-catalog/includes/classes/CatalogCommand.php on line 410 

Deprecated: Optional parameter $args declared before required parameter $opts is implicitly treated as a required parameter in /chroot/var/www/wp-content/plugins/block-catalog/includes/classes/CatalogCommand.php on line 431
```

### Steps to Reproduce

1. With the Block Catalog Plugin active, on a PHP 8.2 environment.
2. Run any WP CLI command like, `wp --help`.
3. You will see a few PHP warnings that originate in the Block Catalog plugin.

### Screenshots, screen recording, code snippet

_No response_

### Environment information

- PHP 8.2+

### WordPress information

- Any

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by dsawardekar**
### Description of the Change

- Upgrade PHPUnit to 9.x
- Fixes failing tests on PHPUnit 9.x
- Fixes PHP warnings when running on PHP 8.2

Closes #33

### How to test the Change

- With the Block Catalog plugin active, on PHP 8.2+
- Run any Block Catalog or WP CLI command like, `wp block-catalog index`
- You should not see any PHP Warnings

### Changelog Entry

Fixed - PHP 8.2 deprecation warnings
Added - PHPUnit 9.x support

### Credits

Props @dsawardekar 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: includes/classes/CatalogBuilder.php [hunk 1]
```diff
@@ -43,10 +43,8 @@ public function catalog( $post_id, $opts = [] ) {
 
 	/**
 	 * Resets the Block Catalog by removing all catalog terms.
-	 *
-	 * @param array $opts Optional opts
 	 */
-	public function delete_index( $opts = [] ) {
+	public function delete_index() {
```
Comment (ravinderk) [inline includes/classes/CatalogBuilder.php]: @dsawardekar Is there any reason to remove this param? If you are refactoring, then can you check compatibility? I notice that few functions are passing  `$opt`.

image

File: includes/classes/CatalogBuilder.php [hunk 2]
```diff
@@ -215,11 +212,10 @@ public function set_post_block_terms( $post_id, $output ) {
 	/**
 	 * Builds a list of Block Term names for a given post.
 	 *
-	 * @param int   $post_id The post id.
-	 * @param array $opts Optional args
+	 * @param int $post_id The post id.
 	 * @return array
 	 */
-	public function get_post_block_terms( $post_id, $opts = [] ) {
+	public function get_post_block_terms( $post_id ) {
```
Comment (ravinderk) [inline includes/classes/CatalogBuilder.php]: @dsawardekar Is there any reason to remove this param? If you are refactoring, then can you check compatibility? I notice that few functions are passing `$opt`.

image

**Review by ravinderk — CHANGES_REQUESTED** [review]
@dsawardekar, I left a few comments. Let me know if you have any questions.

File: includes/classes/CatalogBuilder.php (hunk 2 above)
Comment (dsawardekar) [reply includes/classes/CatalogBuilder.php]: @ravinderk I've added back the extra opts.

File: includes/classes/CatalogBuilder.php (hunk 1 above)
Comment (dsawardekar) [reply includes/classes/CatalogBuilder.php]: @ravinderk I've added back the extra opts.

**Comment by dsawardekar** [comment]
@ravinderk There were a few linter warnings. I've added those opts back with phpcs to ignore the warnings.

**Review by ravinderk — APPROVED** [review]

**Pull request merged**